### PR TITLE
chore(tracing): report peer hostname or IP in dedicated tags

### PR DESCRIPTION
**Pull request by jbertran**
For peer service computation on the backend, we want to provide tags from which to derivate this information in GRPC spans. We can use the value from which we expose the existing `grpc.host` tag:

* If this value is an IP, then we report it in `network.destination.ip`
* If this value is not an IP, then we report it in `peer.hostname`

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [x] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

File: tests/snapshots/tests.contrib.grpc.test_grpc.test_method_service.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
       "grpc.method.service": "Servicer",
       "grpc.status.code": "StatusCode.OK",
       "language": "python",
+      "network.destination.ip": "::",
```
Comment (emmettbutler) [inline tests/snapshots/tests.contrib.grpc.test_grpc.test_method_service.json:23]: This seems like an unexpected value for an IP address field to have. Is it intentional?

Comment (jbertran) [reply tests/snapshots/tests.contrib.grpc.test_grpc.test_method_service.json:23]: `::` is IPv6 loopback range, equivalent to `0.0.0.0` for IPv4, and the [test code that generates the trace](https://github.com/DataDog/dd-trace-py/blob/ba7470328a8f7eccff18bc5e2f71e0c485a98642/tests/contrib/grpc/test_grpc.py#L802-L808) does indeed configure it as IPv6.

File: ddtrace/contrib/grpc/utils.py [hunk 2]
```diff
@@ -43,6 +44,12 @@ def set_grpc_method_meta(span, method, method_kind):
 def set_grpc_client_meta(span, host, port):
     if host:
         span.set_tag_str(constants.GRPC_HOST_KEY, host)
+        try:
+            _ = ipaddress.ip_address(host)
```
Comment (ZStriker19) [inline ddtrace/contrib/grpc/utils.py:48]: Do we need to assign a variable at all here? 
```suggestion
            ipaddress.ip_address(host)
```

**Review by ZStriker19 — APPROVED** [review]
Just a nit/question

**Pull request merged**